### ISSUE TITLE
Improvement request, print modules loaded to stdout.

### ISSUE THREAD

**Issue by spikebike** [hpcugent/easybuild#8]
For 2 reasons I find it very useful to ALWAYS output which module was loaded as they are loaded:
1) users sometimes months or years later want to be able to reproduce scientific results, even more importantly if they are published.  Without know which versions of each dependency were loaded it can be a nightmare to try to figure out which compilers, MPI stack, and libraries were used. 
2) debugging a users problem is quite difficult without knowing exactly what was in a users environment.

I propose that this be appended to every module file:
if [module-info mode load] {
    puts stderr "Module [module-info name] loaded "
}

It has this result:
$ module load GCC
Module GCC/4.6.3 loaded 
$ 

I've never seen this cause a problem, it also raises awareness of the user.  If there are 4 linear algebra libraries available I find it very useful to expose that to the user so that when they hear that a particular library is good/bad at something that they realize they have a choice.

Additionally they don't have to try to figure out from echo $PATH or echo $LD_LIBRARY_PATH, or some flag to try to figure out what version is in use.  

Basically hiding the version info is bad and it should be default and not require adding verbosity flags.
